### DISCUSSION TITLE
[New Feature] TerrainTypes logic enhancement

### DISCUSSION THREAD

**Discussion by McPwny** [Phobos-developers/Phobos#935] [Ideas and feedback]
### Description:
objects in TerrainTypes are handled differently than other object types and have _very_ few ini tags applicable to them. enabling them to, for example, read Palette= and Image=(xxx.shp) in art.ini would effectively make the arcane method of rendering every terrain object in all six theaters obsolete. Additionally, in rules.ini, the tag "IsAnimated=" and its corresponding  "AnimationRate=" and "AnimationProbability=" are hardcoded to the tag "SpawnsTiberium=yes"! de-hardcoding that would allow for animated trees. 

### Additional Files / Links:
 https://modenc.renegadeprojects.com/Theater -->

**Comment by Metadorius** [comment]
Actually we have a WIP branch for tileset custom palettes. Is that what you want partially?

**Comment by McPwny** [comment]
since the tiles interconnect i think its going to be hard on the graphics end to make that feature work but anyways its going to be rather awkward when you do get it working only to realize that you can change the palette of the terrain while the terrain objects are still so strictly hardcoded that you cant even change their palette (nor image).
